### PR TITLE
fix namespace, add create a fork, add forkedfrom field in project ,add tests

### PR DESCRIPTION
**Pull request by EmilienMottet**
- Add a method getNamespace 
- fix URL for Namespace
- create a fork
- add forkedfrom field in project
- add tests
- Use url of Group instead of url of Namespace for transfert, and getNamespaceMembers (A better name would be getGroupMembers)

### CODE REVIEW

File: src/main/java/org/gitlab/api/GitlabAPI.java [hunk 1]
```diff
@@ -935,6 +935,32 @@ public GitlabProject createUserProject(Integer userId, String name, String descr
...
+     * @param namespace The namespace of the fork
+     * @param projectId ProjectId of the project forked
+     * @return The new Gitlab Project
+     * @throws IOException on gitlab api call error
+     */
+    public GitlabProject createFork(String namespace,Integer projectId) throws IOException{
+        Query query = new Query()
+                .appendIf("id", projectId)
+                .append("namespace", namespace);
+        String tailUrl = GitlabProject.URL+"/" + projectId + "/fork";
```
Comment (novalis) [inline src/main/java/org/gitlab/api/GitlabAPI.java]: nit: spaces around +

File: src/main/java/org/gitlab/api/GitlabAPI.java [hunk 2]
```diff
@@ -935,6 +935,32 @@ public GitlabProject createUserProject(Integer userId, String name, String descr
...
+        return dispatch().to(tailUrl,GitlabProject.class);
```
Comment (novalis) [inline src/main/java/org/gitlab/api/GitlabAPI.java]: nit: space after comma

Comment (novalis) [reply src/main/java/org/gitlab/api/GitlabAPI.java]: There are a few more of these elsewhere in the PR; these are just the first two I noticed.

File: src/main/java/org/gitlab/api/models/GitlabProject.java [hunk 3]
```diff
@@ -449,4 +452,27 @@ public String getImportUrl() {
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        GitlabProject that = (GitlabProject) o;
+
+        return id != null ? id.equals(that.id) : that.id == null;
```
Comment (novalis) [inline src/main/java/org/gitlab/api/models/GitlabProject.java]: Is it really right to say that two projects with null id are equal?  Maybe they're two different projects that we have yet to issue create commands for?

Comment (EmilienMottet) [reply src/main/java/org/gitlab/api/models/GitlabProject.java]: you are right

File: src/main/java/org/gitlab/api/GitlabAPI.java (hunk 2 above)
Comment (EmilienMottet) [reply src/main/java/org/gitlab/api/GitlabAPI.java]: I fix it with my ide settings

File: src/main/java/org/gitlab/api/models/GitlabProject.java (hunk 3 above)
Comment (EmilienMottet) [reply src/main/java/org/gitlab/api/models/GitlabProject.java]: @novalis 
If id is null so the 2 projects are not equals ? or It need to compare project name + project namespace ?
Thank you for your advice.

Comment (novalis) [reply src/main/java/org/gitlab/api/models/GitlabProject.java]: I like the idea of comparing project name and namespace.  I think I might do that in all cases -- not just if the ids are null.

File: src/main/java/org/gitlab/api/GitlabAPI.java [hunk 4]
```diff
@@ -674,10 +674,11 @@ public GitlabProject getProject(String namespace, String projectName) throws IOE
     /**
      * Get a list of the namespaces of the authenticated user.
      * If the user is an administrator, a list of all namespaces in the GitLab instance is shown.
+     *
```
Comment (novalis) [inline src/main/java/org/gitlab/api/GitlabAPI.java]: It's generally better to edit these into the original commit(s) using git rebase.

Comment (EmilienMottet) [reply src/main/java/org/gitlab/api/GitlabAPI.java]: yes, I will squash it.

File: src/main/java/org/gitlab/api/GitlabAPI.java [hunk 5]
```diff
@@ -2179,7 +2215,7 @@ public void deleteProjectMember(Integer projectId, Integer userId) throws IOExce
      * @throws IOException on gitlab api call error
      */
     public List<GitlabProjectMember> getNamespaceMembers(Integer namespaceId) throws IOException {
-        String tailUrl = GitlabNamespace.URL + "/" + namespaceId + GitlabProjectMember.URL;
+        String tailUrl = GitlabGroup.URL + "/" + namespaceId + GitlabProjectMember.URL;
```
Comment (timols) [inline src/main/java/org/gitlab/api/GitlabAPI.java]: Should there be new methods such as `getGroupMembers` etc? Do you think it might cause confusion?

Comment (cpfeiffer) [reply src/main/java/org/gitlab/api/GitlabAPI.java]: The API doc already states that the methods fails when the namespace is not a group. I don't know how exactly it fails. but renaming the method to getGroupMembers() would make its purpose clear.

**Comment by Spenhouet** [comment]
This pull request is now over 3 months old. When will this be merged and released?

**Pull request merged**